src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -569,10 +581,8 @@ case class WorkflowActor(workflow: WorkflowDescriptor, backend: Backend)
       // Running and Aborting have explicit handlers for Running and no pending writes, so either we are not in those
       // states or the specified key is no longer Running and/or has pending writes.
       stay()
-    case Event(PersistenceCompleted(callKey, status, _), data) =>
-      logger.debug(s"Got whenUnhandled PersistenceCompleted(${callKey.tag}): $status")
-      val updatedData = data.removePersisting(callKey, status)
-      stay using updatedData
+    case Event(m: PersistenceCompleted, data) => removePendingCallKeyPersistence(data, m)
+    case Event(m: PersistenceFailed, data) => removePendingCallKeyPersistence(data, m)
```

Comment:
if everything is under the base trait then these two and the `PersistencesCompleted` handler could be replaced with

```scala
case Event(m: PersistenceCompleted, data) => removePendingCallKeyPersistence(data, m)
```